vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java

```diff
@@ -414,9 +415,14 @@ protected Stack initializeContents() {
       if (configurable instanceof PieceSlot) {
         PieceSlot slot = (PieceSlot) configurable;
         GamePiece p = slot.getPiece();
-        p = PieceCloner.getInstance().clonePiece(p);
-        GameModule.getGameModule().getGameState().addPiece(p);
-        s.add(p);
+        if (p != null) { // In case slot fails to "build the piece", which is a possibility.
+          p = PieceCloner.getInstance().clonePiece(p);
+          GameModule.getGameModule().getGameState().addPiece(p);
+          s.add(p);
+        } 
+        else {
+          Chatter.warning("- Bad Piece Data - Failed to create piece in At-Start Stack: " + getConfigureName());
+        }
```

Comment:
Rather than run this directly into Chatter.warning, it might be better to use the existing structures and do this:

ErrorDialog.dataError(new BadDataReport(this, message, data));

This particular issue is a bad PieceDefinition in a PieceSlot, so you could add a new Constructor to BadDataReport(PieceSlot slot, String message, String data)  and then just do

ErrorDialog.dataError(new BadDataReport(slot, "Piece Definition failed to build", "Attempting to build piece "+num+" in At-Start stack"+getConfigureName());

The BadDataReport would handle the actual formatting, This could probably be used anywhere the PieceCloner fails to build a piece.

Have a Play, might be better to sub-class BadDataReport into BadPieceDefintionReport, BadTraitReport, BadOtherThingReport. etc. That would allow you to tailor the Constructors better. The Trait stuff in BadDataReport should really be in a subclass BadTraitReport. 

This will also give us a central place to change how we handle these.